src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -3333,7 +3333,7 @@ void KeyCondition::prepareBloomFilterData(std::function<std::optional<size_t>(si
             KeyCondition::BloomFilterData::HashesForColumns::value_type hashes_for_column;
             hashes_for_column.emplace_back(*hashed_value);
 
-            hashes.emplace_back(std::move(hashes_for_column));
+            hashes.emplace_back(static_cast<std::vector<uint64_t>>(std::move(hashes_for_column)));
```

Comment:
Would just `hashes.emplace_back({*hashed_value});` work?

I'd use uint64_t for the hash everywhere in this PR (changing hash_one and hash_many return types), since that's what the arrow's functions return, and all hashes in this PR come from those functions.